### ISSUE TITLE
Cannot Compile Latest Relion with Cuda OpenMPI 2.1.0 and GCC 6.1.0

### ISSUE THREAD

**Issue by JusticeForMikeBrown** [3dem/relion#230]
Hi!  We are running on a new cluster with Tesla P100s.  We are getting much better results on a single node and now want to use MPI across the cluster.  I cannot get relion latest to compile.  The below error happens with fftw-devel 3.3.3 from Centos repos as well as the external v 3.3.4 build which comes with Relion.  In addition I disabled fltk to remove it as a variable.

Can you please help?

```
node007[/cm/shared/scratch/relion/build]$ gcc --version
gcc (GCC) 6.1.0
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
node007[/cm/shared/scratch/relion/build]$ /cm/shared/apps/openmpi/cuda/64/2.1.0/bin/mpicxx --version                                                                                                          
g++ (GCC) 6.1.0
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
node007[/cm/shared/scratch/relion/build]$ cmake -DGUI=OFF .. -DCMAKE_INSTALL_PREFIX=/cm/shared/scratch/relion -DCMAKE_BUILD_TYPE=Benchmarking 

-- Omitting GUI targets as per your request
-- MPI_INCLUDE_PATH : /cm/shared/apps/openmpi/cuda/64/2.1.0/include
-- MPI_LIBRARIES : /cm/shared/apps/openmpi/cuda/64/2.1.0/lib/libmpi.so
-- MPI_CXX_INCLUDE_PATH : /cm/shared/apps/openmpi/cuda/64/2.1.0/include
-- MPI_CXX_LIBRARIES : /cm/shared/apps/openmpi/cuda/64/2.1.0/lib/libmpi.so
-- Found FFTW: fftw3
-- FFTW_LIBRARIES: /usr/lib64/libfftw3.so
-- Building shared libs (smaller build size and binaries)
-- CMAKE_BINARY_DIR:/cm/shared/scratch/relion/build
-- added autopick...
```
Make Error:

```
[ 41%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/particle_sorter_mpi.cpp.o
[ 42%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/pipeliner.cpp.o
[ 42%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/postprocessing.cpp.o
[ 43%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/preprocessing.cpp.o
[ 44%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/preprocessing_mpi.cpp.o
[ 45%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/projector.cpp.o
[ 46%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/strings.cpp.o
[ 47%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/symmetries.cpp.o
[ 48%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/tabfuncs.cpp.o
[ 48%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/time.cpp.o
[ 49%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/transformations.cpp.o
[ 50%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/Healpix_2.15a/cxxutils.cc.o
[ 51%] Building CXX object src/apps/CMakeFiles/relion_lib.dir/__/Healpix_2.15a/healpix_base.cc.o
[ 52%] Linking CXX shared library ../../lib/librelion_lib.so
[ 52%] Built target relion_lib
[ 53%] Building CXX object src/apps/CMakeFiles/star_compare.dir/star_compare.cpp.o
[ 54%] Linking CXX executable ../../bin/relion_star_compare
../../lib/librelion_gpu_util.so: undefined reference to `FileName::compose(std::string const&, long, std::string const&, int)'
../../lib/librelion_gpu_util.so: undefined reference to `FileName::addExtension(std::string const&) const'
../../lib/librelion_gpu_util.so: undefined reference to `FileName::beforeLastOf(std::string const&) const'
../../lib/librelion_gpu_util.so: undefined reference to `EMDL::label2Str(EMDLabel const&)'
../../lib/librelion_gpu_util.so: undefined reference to `floatToString(float, int, int)'
../../lib/librelion_gpu_util.so: undefined reference to `FileName::decompose(long&, std::string&) const'
../../lib/librelion_gpu_util.so: undefined reference to `FileName::contains(std::string const&) const'
../../lib/librelion_gpu_util.so: undefined reference to `MetaDataContainer::getValue(EMDLabel, std::string&)'
../../lib/librelion_gpu_util.so: undefined reference to `RelionError::RelionError(std::string const&, std::string const&, long)'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/relion_star_compare] Error 1
make[1]: *** [src/apps/CMakeFiles/star_compare.dir/all] Error 2
make: *** [all] Error 2

```

**Comment by bforsbe** [comment]
gcc 6 is not supported by cuda 8,  as far as i know, and there has also been similar reports which were avoided by using cmake less than 3.7. Which cmak-version are you using?

**Comment by david-barker** [comment]
I'm hitting this same issue; gcc 5.4, CUDA 7.5 & cmake 3.6.2

**Comment by bforsbe** [comment]
I don't think your config step is picking up everything you want it to. On my system the cmake (3.6.3) config picks up cuda 7.4 ang gcc 5.4, and shouts loudly and immediately that this is unsupported; 

```
$ cmake -DCUDA_ARCH=52 ..
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
...
-- Found CUDA: /opt/tcbsys/cuda/7.5 (found version "7.5") 
-- Using cuda wrapper to compile....
-- Cuda version is >= 7.5 and single-precision build, enable double usage warning.
...
$ make -j8
Scanning dependencies of target copy_scripts
[  1%] Building NVCC (Device) object src/apps/CMakeFiles/relion_gpu_util.dir/__/gpu_utils/cuda_kernels/relion_gpu_util_generated_helper.cu.o
[  2%] Building NVCC (Device) object src/apps/CMakeFiles/relion_gpu_util.dir/__/gpu_utils/relion_gpu_util_generated_cuda_autopicker.cu.o
...
[  7%] Building NVCC (Device) object src/apps/CMakeFiles/relion_gpu_util.dir/__/gpu_utils/relion_gpu_util_generated_cuda_backprojector.cu.o
In file included from /opt/tcbsys/cuda/7.5/include/cuda_runtime.h:76:0,
                 from <command-line>:0:
/opt/tcbsys/cuda/7.5/include/host_config.h:115:2: error: #error -- unsupported GNU version! gcc versions later than 4.9 are not supported!
 #error -- unsupported GNU version! gcc versions later than 4.9 are not supported!
```

You might be using some other combination of program versions if cmake finds thing in a different order from what your path specifies.

**Comment by bforsbe** [comment]
Closing due to use of unsupported version combinations (cuda 8 supports gcc 5.3 at most, cuda 7.5 even less so)
